apps/api/v2/src/modules/stripe/controllers/stripe.controller.ts

```diff
@@ -85,18 +88,46 @@ export class StripeController {
     @Query("error") error: string | undefined,
     @Query("error_description") error_description: string | undefined
   ): Promise<StripCredentialsSaveOutputResponseDto> {
-    const accessToken = JSON.parse(state).accessToken;
-
-    // user cancels flow
-    if (error === "access_denied") {
-      return { url: getOnErrorReturnToValueFromQueryState(state) };
+    if (!state) {
```

Comment:
The code from line 95 to 130 should be moved to stripe service - it's too much code for a controller - controllers should only extract information, pass it to service and then return value of the service.